Source/Core/Core/ARDecrypt.cpp

```diff
@@ -176,40 +179,35 @@ static void generateseeds(u32* seeds, const u8* seedtable, u8 doreverse)
 
       array1[j] = array0[tmp];
     }
-    for (int j = 0; j < 0x30; j++)
+    for (u32 j = 0; j < 0x30; j++)
     {
-      if (!array1[gentable3[j] - 1])
+      if (array1[gentable3[j] - 1] == 0)
       {
         continue;
       }
-      tmp = (((j * 0x2AAB) >> 16) - (j >> 0x1F));
+
+      const u8 tmp = (((j * 0x2AAB) >> 16) - (j >> 0x1F));
       array2[tmp] |= (gentable1[j - (tmp * 6)] >> 2);
     }
     seeds[i << 1] = ((array2[0] << 24) | (array2[2] << 16) | (array2[4] << 8) | array2[6]);
     seeds[(i << 1) + 1] = ((array2[1] << 24) | (array2[3] << 16) | (array2[5] << 8) | array2[7]);
   }
 
-  if (!doreverse)
+  int j = 0x1F;
+  for (int i = 0; i < 16; i += 2)
   {
-    int j = 0x1F;
-    for (int i = 0; i < 16; i += 2)
-    {
-      tmp3 = seeds[i];
-      seeds[i] = seeds[j - 1];
-      seeds[j - 1] = tmp3;
-
-      tmp3 = seeds[i + 1];
-      seeds[i + 1] = seeds[j];
-      seeds[j] = tmp3;
-      j -= 2;
-    }
+    u32 tmp3 = seeds[i];
```

Comment:
Possible followup PR: use std::swap to get rid of `tmp3`?

Reply:
Unfortunately std::swap isn't constexpr in C++17, so this can't be done. This would be a candidate whenever we move to C++20 though.